provider/identity_provider_resource.go

```diff
@@ -129,8 +130,20 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 		return
 	}
 
-	// Wait till the cluster is ready:
 	resource := r.collection.Cluster(state.Cluster.Value)
+	// We expect the cluster to be already exist
+	// Try to get it and if result with NotFound error, return error to user
+	if resp, err := resource.Get().SendContext(ctx); err != nil && resp.Status() == 404 {
+		message := fmt.Sprintf("Cluster %s not found, error: %v", state.Cluster.Value, err)
+		tflog.Error(ctx, message)
+		response.Diagnostics.AddError(
+			"Can't poll cluster state",
+			message,
+		)
+		return
+	}
```

Comment:
```suggestion
	if resp, err := resource.Get().SendContext(ctx); err != nil {
	        if resp.Status() == 404 {
	        		message := fmt.Sprintf("Cluster %s not found, error: %v", state.Cluster.Value, err)
			        tflog.Error(ctx, message)
	        		
	        } else {
	        	        message := fmt.Sprintf("Can't poll cluster with ID %s, error %v", state.ID.Value, err)
	        }
	        response.Diagnostics.AddError(
	        	        "Can't poll cluster state",
	        	        message,
	        	        )
		return
	}
```

Reply:
I think it has to fail only if the cluster is not found, for any other error (like possible connectivity issue) better to wait, no?
The point here is to fail in case the cluster doesn't exist